Tribler/Core/Modules/tracker_manager.py

```diff
@@ -95,6 +95,11 @@ def update_tracker_info(self, tracker_url, is_successful):
         :param tracker_url: The given tracker_url.
         :param is_successful: If the check was successful.
         """
+        # It is possible that a Deferred call could not be cancelled and invokes this function while
+        # we are shutting down, in that case we ignore it.
+        if not self._tracker_dict:
```

Comment:
I have investigated the torrent_checker where the calls come from thoroughly and everything seems good there. Every time a session is created it is added to the list, and on `shutdown` a loop runs over the list calling the `cleanup` for every session, which cancels the deferred. So it is really weird that it is firing. Also, the torrent checker shutdown happens before the tracker manager in launchmanycores.

I have removed the redundant code btw.